src/LanguageServer/Test/SymbolIndexTests.cs

```diff
@@ -0,0 +1,185 @@
+            symbols.Should().BeEquivalentToWithStrictOrdering(new[] {
+                new HierarchicalSymbol("y", SymbolKind.Variable, new SourceSpan(1, 1, 1, 2)),
+            });
+        }
+
+        [TestMethod, Priority(0)]
+        public async Task IndexHierarchicalDocumentNotFoundAsync() {
+            ISymbolIndex index = MakeSymbolIndex();
+            var path = TestData.GetDefaultModulePath();
+
+            var symbols = await index.HierarchicalDocumentSymbolsAsync(path);
+            symbols.Should().BeEmpty();
+        }
+
+        [TestMethod, Priority(0)]
+        public async Task IndexWorkspaceSymbolsFlattenAsync() {
+            var code = @"class Foo(object):
+    def foo(self, x): ...";
+
+            ISymbolIndex index = MakeSymbolIndex();
+            var path = TestData.GetDefaultModulePath();
+
+            index.Add(path, DocumentWithAst(code));
+
+            var symbols = await index.WorkspaceSymbolsAsync("", maxSymbols);
+            symbols.Should().BeEquivalentToWithStrictOrdering(new[] {
+                new FlatSymbol("Foo", SymbolKind.Class, path, new SourceSpan(1, 7, 1, 10)),
+                new FlatSymbol("foo", SymbolKind.Method, path, new SourceSpan(2, 9, 2, 12), "Foo"),
+                new FlatSymbol("self", SymbolKind.Variable, path, new SourceSpan(2, 13, 2, 17), "foo"),
+                new FlatSymbol("x", SymbolKind.Variable, path, new SourceSpan(2, 19, 2, 20), "foo"),
+            });
+        }
+
+        [TestMethod, Priority(0)]
+        public async Task IndexWorkspaceSymbolsFilteredAsync() {
+            var code = @"class Foo(object):
+    def foo(self, x): ...";
+
+            ISymbolIndex index = MakeSymbolIndex();
+            var path = TestData.GetDefaultModulePath();
+
+            index.Add(path, DocumentWithAst(code));
+
+            var symbols = await index.WorkspaceSymbolsAsync("x", maxSymbols);
+            symbols.Should().BeEquivalentToWithStrictOrdering(new[] {
+                new FlatSymbol("x", SymbolKind.Variable, path, new SourceSpan(2, 19, 2, 20), "foo"),
+            });
+        }
+
+        [TestMethod, Priority(0)]
+        public async Task IndexWorkspaceSymbolsNotFoundAsync() {
+            ISymbolIndex index = MakeSymbolIndex();
+            var path = TestData.GetDefaultModulePath();
+
+            var symbols = await index.WorkspaceSymbolsAsync("", maxSymbols);
+            symbols.Should().BeEmpty();
+        }
+
+        [TestMethod, Priority(0)]
+        public async Task IndexWorkspaceSymbolsCaseInsensitiveAsync() {
+            var code = @"class Foo(object):
+    def foo(self, x): ...";
+
+            ISymbolIndex index = MakeSymbolIndex();
+            var path = TestData.GetDefaultModulePath();
+
+            index.Add(path, DocumentWithAst(code));
+
+            var symbols = await index.WorkspaceSymbolsAsync("foo", maxSymbols);
+            symbols.Should().BeEquivalentToWithStrictOrdering(new[] {
```

Comment:
`symbols.Should().Equal(`

Reply:
This is my code from #521. Does `Equal` actually work the same?

Reply:
Yep, you're right - `Equal` won't work because `FlatSymbol` doesn't override `Equals` and it isn't a struct, so `Equal` will just compare the references, while `BeEquivalentTo` will compare all the public properties and fields.